packages/edit-navigation/src/components/name-editor/index.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useEffect, useRef, useState, useContext } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import { TextControl } from '@wordpress/components';
+import {
+	IsMenuEditorFocused,
+	useMenuEntity,
+	useNavigationEditorMenu,
+} from '../../hooks';
+import { useInstanceId } from '@wordpress/compose';
+
+export function NameEditor() {
+	const [ isMenuNameEditFocused, setIsMenuNameEditFocused ] = useContext(
+		IsMenuEditorFocused
+	);
+
+	const { menuName, menuId } = useNavigationEditorMenu();
+	const { editMenuName } = useMenuEntity( menuId );
+	const inputRef = useRef();
+	const [ tmpMenuName, setTmpMenuName ] = useState( menuName );
```

Comment:
This state doesn't seem like it should be needed. It should be possible to get the current value from the edited entity record. Feels like something `useMenuEntity` should be able to return.